packages/node/src/core/evm/providers.ts

```diff
@@ -0,0 +1,113 @@
+import { ethers } from 'ethers';
+import { Airnode, Convenience } from './contracts';
+import { go, retryOperation } from '../utils/promise-utils';
+import * as logger from '../logger';
+import * as wallet from './wallet';
+import { LogsData } from '../../types';
+
+interface BaseFetchOptions {
+  airnodeAddress: string;
+  convenienceAddress: string;
+  provider: ethers.providers.JsonRpcProvider;
+}
+
+interface FindOptions extends BaseFetchOptions {
+  providerId: string;
+}
+
+interface CreateOptions extends BaseFetchOptions {
+  adminAddress: string;
+  airnodeAddress: string;
+  provider: ethers.providers.JsonRpcProvider;
+  xpub: string;
+}
+
+interface ProviderWithBlockNumber {
+  adminAddress: string;
+  blockNumber: number;
+  providerExists: boolean;
+  xpub: string;
+}
+
+export async function findWithBlock(fetchOptions: FindOptions): Promise<LogsData<ProviderWithBlockNumber | null>> {
+  const convenience = new ethers.Contract(fetchOptions.convenienceAddress, Convenience.ABI, fetchOptions.provider);
+  const contractCall = () => convenience.getProviderAndBlockNumber(fetchOptions.providerId);
+  const retryableContractCall = retryOperation(2, contractCall, { timeouts: [4000, 4000] }) as Promise<any>;
+
+  const fetchLog = logger.pend('INFO', 'Fetching current block and provider admin details...');
+
+  const [err, res] = await go(retryableContractCall);
+  if (err || !res) {
+    const errLog = logger.pend('ERROR', 'Unable to fetch current block and provider admin details', err);
+    return [[fetchLog, errLog], null];
+  }
+
+  const data: ProviderWithBlockNumber = {
+    adminAddress: res.admin,
+    // Converting this BigNumber to a JS number should not throw as the current block number
+    // should always be a valid number
+    blockNumber: res.blockNumber.toNumber(),
+    providerExists: !!res.xpub && res.xpub !== '',
+    xpub: res.xpub,
+  };
+
+  const blockLog = logger.pend('INFO', `Current block: ${res.blockNumber}`);
+
+  if (!data.providerExists) {
+    const providerLog = logger.pend('INFO', 'EVM provider not found');
+    const logs = [fetchLog, blockLog, providerLog];
+    return [logs, data];
+  }
+
+  const addressLog = logger.pend('INFO', `Admin address: ${res.admin}`);
+  const xpubLog = logger.pend('INFO', `Admin extended public key: ${res.xpub}`);
```

Comment:
Admin extended public key -> Provider extended public key

There are two separate entities:
The node (or provider), which has a private key, its mnemonic, xpub, designatedWallets, etc.
The provider admin, which is an Ethereum account that is probably controlled by some employee of the API provider, which has weaker credentials compared to the node itself.

Reply:
Thanks, you're right. I didn't think too much about these log messages. There are so many uses of the word "provider", it all gets quite confusing 😅